lib/OpenQA/WebAPI/Controller/API/V1/Job.pm

```diff
@@ -124,6 +125,11 @@ sub list ($self) {
     my $latest = $validation->param('latest');
     my $schema = $self->schema;
     my $rs = $schema->resultset('Jobs')->complex_query(%args);
+
+    if (defined(my $not_groupid = $self->param('not_groupid'))) {
```

Comment:
Not sure how other parameters are handled here but this should probably use the value from the validation object:
```suggestion
    if (defined(my $not_groupid = $validation->param('not_groupid'))) {
```